packages/flutter/test/gestures/force_press_test.dart

```diff
@@ -606,4 +606,16 @@ void main() {
     expect(updated, 1);
     expect(ended, 1);
   });
+
+  testWidgets('ForecePressGestureRecognizer asserts when kind and supportedDevices are both set', (WidgetTester tester) async {
+    try {
```

Comment:
This pattern seems unsafe. Consider if `ForcePressGestureRecognizer` accidentally had the asset deleted. Then there would be no error to catch and the test would just pass as is.

consider:
```dart
expect(() => ForcePressGestureRecognizer(
    kind: PointerDeviceKind.touch,
    supportedDevices: <PointerDeviceKind>{ PointerDeviceKind.touch },
  ), throwsA(isA<AssertionError>().having((e) => e.toString(), 'description',  contains('kind == null || supportedDevices == null'))
```

Reply:
Ah I forgot to add the fail state at the end of the try statement. I like yours better. Maybe we should update other tests that are like this, I've forgotten the fail state before, and seen others do the same. Thanks! 🙏

Reply:
@a14n Is this pattern lintable to suggest the `throwsA` pattern over the error-prone try/catch pattern in tests?

Reply:
Running such a lint on flutter doesn't detect any similar case. So I don't think it's common enough to justify the existence/maintenance of this lint.

Reply:
My bad, I made a mistake in my test. There are actually cases where it fails.

Reply:
Actually I found several cases (a PR will be sent soon) but there are false positives that are hard to detect. So the lint should not be that useful.